### PR TITLE
Fix: docs

### PR DESCRIPTION
**Pull request by prithviraj-gotepatil-algobulls**
- updated docs for 
  - new format for strategy
  - new parameter key names for backtesting and papertrading
  - quantstats mode for statistics report generation
  - new flowchart

### CODE REVIEW

File: docs/home.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Pyalgotrading is an official Python package that offers algorithmic trading APIs powered by [AlgoBulls](https://www.algobulls.com/). It provides support for creating and uploading strategies on the [AlgoBulls](https://www.algobulls.com/) platform, as well as a pool of free strategies at [pyalgostrategypool](https://github.com/algobulls/pyalgostrategypool). It
+also supports over 150 technical indicators, multiple candlestick patterns and intervals, and different types of orders. Pyalgotrading allows for backtesting, paper trading, and real trading (subject to approval), and supports multiple brokers for
+real trading. Real-time logs and reports are available for all types of trading, and candlestick charts can be plotted using [plotly.py](https://github.com/plotly/plotly.py).
+
+---
+
+## [PythonBuild](python_build/index.md)
+
+---
+The PythonBuild service allows users to code their own trading strategy from scratch or modify templates. Users can access the [AlgoBulls](https://www.algobulls.com/) Core Trading Engine and historical data to gauge strategy performance. Users can tweak parameters and view
```
Comment (akhil-jain-algobulls) [inline docs/home.md:17]: `or modify templates`

need good clarity and coverage on templates. Maybe one or two additional lines can highlight this

Comment (akhil-jain-algobulls) [reply docs/home.md:17]: ` and view visual analytics to improve performance`

this should be something along the lines of
` and use the visual analytics to gauge and improve the performance of the strategy`

Comment (akhil-jain-algobulls) [reply docs/home.md:17]: feel free to improve both lines @prithviraj-gotepatil-algobulls 

you can again use generative and/or summarization ai tools if you have done so before

File: docs/pyalgotrad/index.md [hunk 2]
```diff
@@ -11,76 +10,56 @@
...
-
+Package can be easily installed using `pip` - 
 ```
 pip install pyalgotrading
 ```
 
 ### Support / Getting Help
-
 - *Bug Reporting / New Feature Request*: Please [create a new issue](https://github.com/algobulls/pyalgotrading/issues/new) here on GitHub.
 - *Discussion Forum*: Coming Soon!
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:59]: we can add
`https://community.algobulls.com/c/github/pyalgotrading/42`
for the discussion forum till we get something better

File: docs/pyalgotrad/index.md [hunk 3]
```diff
@@ -11,76 +10,56 @@
...
 Backtesting, Paper Trading and Real Trading can be performed on the same strategy code base!
 
 ### Documentation
-
 You can find the docs [here](https://algobulls.github.io/pyalgotrading/).
 
-### Python
-
+### Python 
 - Python Support: `Python 3.6+`.
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:47]: python version should be 3.10

File: docs/pyalgotrad/index.md [hunk 4]
```diff
@@ -11,76 +10,56 @@
...
     - Statistics of (PnL report)
     - Order History for each order with state transitions & timestamps
 - Plot Candlestick charts using [plotly.py](https://github.com/plotly/plotly.py)
-
+    
 Backtesting, Paper Trading and Real Trading can be performed on the same strategy code base!
 
 ### Documentation
-
 You can find the docs [here](https://algobulls.github.io/pyalgotrading/).
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:44]: do we need a section where the docs are pointing to themselves?

File: docs/pyalgotrad/index.md [hunk 5]
```diff
@@ -11,76 +10,56 @@
...
     - Support for **Regular Orders** and **Bracket Orders**
     - Support for **MARKET**, **LIMIT**, **STOPLOSS**, **STOPLOSS-LIMIT** orders
     - Support for **INTRADAY** and **DELIVERY** orders
 - Support for **Backtesting**
 - Support for **Paper Trading**
 - Support for **Real Trading**
-- Support for multiple brokers for Real Trading
-    - Currently supported brokers - Zerodha
+- Support for multiple brokers for Real Trading 
+    - Currently supported brokers - Zerodha
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:31]: `Currently supported brokers - Click Here `

You can point `Click Here` to `https://app.algobulls.com/user/brokerlogin`

Comment (akhil-jain-algobulls) [reply docs/pyalgotrad/index.md:31]: remove the below 2 lines

File: docs/pyalgotrad/index.md [hunk 6]
```diff
@@ -11,76 +10,56 @@
...
-
 - Powered by the [AlgoBulls Platform](https://algobulls.com)
 - Everything related to Algorithmic Trading Strategies!
     - Create & upload strategies on the [AlgoBulls Platform](https://algobulls.com)
     - Free pool of Strategies are available separately at [pyalgostrategypool](https://github.com/algobulls/pyalgostrategypool)!
     - Support for all 150+ Technical Indicators provided by [TA-Lib](https://mrjbq7.github.io/ta-lib/index.html)
     - Support for multiple candlesticks patterns - Japanese OHLC, Renko, Heikin-Ashi, Linebreak
-    - Support for multiple candle inntervals - 1 minute, 3 minutes, 5 minutes, 10 minutes, 15 minutes, 30 minutes, 1 hour, 1 day.
+    - Support for multiple candle intervals - 1 minute, 3 minutes, 5 minutes, 10 minutes, 15 minutes, 30 minutes, 1 hour, 1 day. 
     - Support for **Regular Orders** and **Bracket Orders**
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:24]: `Support for **Regular Orders** and **Bracket Orders**`

remove this line

File: docs/pyalgotrad/index.md [hunk 7]
```diff
@@ -11,76 +10,56 @@
...
     - Support for **MARKET**, **LIMIT**, **STOPLOSS**, **STOPLOSS-LIMIT** orders
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:25]: `- Support for **MARKET**, **LIMIT**, **STOPLOSS**, **STOPLOSS-LIMIT** orders`

change this to

`- Support for **MARKET**, **LIMIT**, **STOPLOSS-MARKET**, **STOPLOSS-LIMIT** orders`

File: docs/pyalgotrad/index.md [hunk 8]
```diff
@@ -11,76 +10,56 @@
...
     - Support for **INTRADAY** and **DELIVERY** orders
 - Support for **Backtesting**
 - Support for **Paper Trading**
 - Support for **Real Trading**
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:29]: - Support for **Backtesting**
- Support for **Paper Trading**
- Support for **Real Trading**

change this into one line

- Support for **Backtesting** and  **Paper Trading**

File: docs/pyalgotrad/index.md [hunk 9]
```diff
@@ -11,76 +10,56 @@
...
-- Support for multiple brokers for Real Trading
-    - Currently supported brokers - Zerodha
+- Support for multiple brokers for Real Trading 
+    - Currently supported brokers - Zerodha 
     - Coming soon - AliceBlue, Mastertrust
     - More on the way!
 - Real-time Logs for Backtesting, Paper Trading, Real Trading
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:34]: remove `Real Trading` from everywhere

Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 8 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 7 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 6 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 5 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 4 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 3 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/index.md (hunk 2 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/home.md (hunk 1 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/home.md]: done

File: docs/pyalgotrad/introduction.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+This tutorial aims to help users practice and get familiar with the [Pyalgotrading](https://github.com/algobulls/pyalgotrading/) package, which is a Python-based algorithmic trading library. It allows you to create and execute your trading strategies on various financial instruments such as equities, futures, options, and more. To follow this tutorial, you will need to have an active and working [AlgoBulls](https://www.algobulls.com/) account, which is a platform for creating and testing trading strategies.
+
+---
+## [Prerequisites](prerequisites.md)
+
+Before starting the tutorial, ensure that you have the following setup:
+- An active and working [AlgoBulls](https://www.algobulls.com/) account
+- Python IDE of your choice (if you want to code your own strategy)
+- Jupyter Notebook (for points 2,3, and 4)
+
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/introduction.md:14]: the prerequisites above are all visible on the same line. they are separate points and should be on their own each line

Comment (akhil-jain-algobulls) [reply docs/pyalgotrad/introduction.md:14]: `(for points 2,3, and 4)`

check and add which part is relevant to point 1 also

File: docs/pyalgotrad/structure.md [hunk 11]
```diff
@@ -1,50 +1,60 @@
...
 ---
+#### Supporting Methods
+
 You have the following supporting methods:
 
-1. `__init__()` - This is the constructor method used in every strategy. You can perform sanity checks here for each parameter passed as an argument to the strategy.
-   You can also declare the set of variables that you may need for your strategy.
-2. `initialize()` - This method is called every time the strategy starts for the first time as well as the start of every trading day.
-   You may set initial values for the set of variables that you have declared in the previous method.
+1. `__init__()` - Thia ia the constructor method used in every strategy. You can perform sanity checks here for each parameter passed as an argument to the strategy.
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/structure.md:36]: `Thia ia the`

fix typo

File: docs/pyalgotrad/codeguide.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+!!! Note
+    * Pandas is available with limited functionality. Direct printing of dataframes is disabled.
+    * It is a good idea to include the version number while installing a package using `pip`.
+    * Example - `pip install numpy==1.16.3` 
+
+You may get the following message if the strategy fails to adhere to the guidelines.
+
+```
+Strategy breached code of conduct. Please adjust your strategy to follow guidelines and run again. Halting strategy...
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/codeguide.md:30]: does this come from pyalgotrading or core?

If it comes from pyalgotrading, just check if this print is updated in any way

If it is, update it here too

File: docs/pyalgotrad/index.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    - Free pool of Strategies are available separately at [pyalgostrategypool](https://github.com/algobulls/pyalgostrategypool)!
+    - Support for all 150+ Technical Indicators provided by [TA-Lib](https://mrjbq7.github.io/ta-lib/index.html)
+    - Support for multiple candlesticks patterns - Japanese OHLC, Renko, Heikin-Ashi, Linebreak
+    - Support for multiple candle intervals - 1 minute, 3 minutes, 5 minutes, 10 minutes, 15 minutes, 30 minutes, 1 hour, 1 day. 
+    - Support for **MARKET**, **LIMIT**, **STOPLOSS-MARKET**, **STOPLOSS-LIMIT** orders
+    - Support for **INTRADAY** and **DELIVERY** orders
+- Support for **Backtesting** and **Paper Trading**
+
+    - Currently supported [brokers](https://app.algobulls.com/user/brokerlogin) 
+- Real-time Logs for Backtesting and Paper Trading
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:28]: @pushpakdagade I told @prithviraj-gotepatil-algobulls to remove RT from everywhere

we are planning to support RT through pyalgotrading?

File: docs/pyalgotrad/index.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    - Statistics of (PnL report)
+    - Order History for each order with state transitions & timestamps
+- Plot Candlestick charts using [plotly.py](https://github.com/plotly/plotly.py)
+    
+Backtesting and Paper Trading can be performed on the same strategy code base!
+
+### Python 
+- Python Support: `Python 3.10+`.
+- Python Requirements: See [requirements.txt](https://github.com/algobulls/pyalgotrading/blob/master/requirements.txt).
+- We recommend you to use the latest version of Python (v3.8+) to enjoy better performance benefits, especially for pandas. (For Python 3.6, the latest supported Pandas version is v0.25.3. For Python 3.8, the latest supported version is v1.0.x)
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/index.md:40]: update python and pandas versions'
I think we remove the bracket and its contents

**Comment by pushpakdagade** [comment]
Yes, for certain exchanges.

File: docs/pyalgotrad/prerequisites.md [hunk 15]
```diff
@@ -1,21 +1,38 @@
...
-2. **Python 3.7+ ([Link](https://www.python.org/downloads/))** - The AlgoBulls platform can run all strategies that conform to Python 3.7 and above.
-3. **A Python IDE of your choice, like Pycharm ([Link](https://www.jetbrains.com/pycharm/download/))** - As mentioned in the [Introduction], this tutorial will demonstrate the pyalgotrading package by implementing the Simple Moving Average Crossover
-   Strategy for Regular Orders. You will make this strategy as a separate python file, and the PyCharm Community Edition IDE is enough for this purpose. If you are already using a Python IDE of your choice, you may skip this step.
+1. **AlgoBulls Account ([Link](https://www.algobulls.com/))** - You will need this account to upload and test your strategies. The account needs to be setup with the proper risk management values and a bound broker before any trades can be executed. Make sure the account is properly setup or contact [AlgoBulls Support](mailto:support@algobulls.com) for any help.
+
+    ---
+
+2. **Python 3.7+ ([Link](https://www.python.org/downloads/))** - The AlgoBulls platform can run all strategies that conform to Python 3.7 and above.
```
Comment (akhil-jain-algobulls) [inline docs/pyalgotrad/prerequisites.md:10]: update python versions

File: docs/pyalgotrad/codeguide.md [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+
+## List of Guidelines
+---
+1. The following Python packages/modules are available to you:
+
+    | Package / Module                                           | Version  | 
+    |------------------------------------------------------------|----------| 
+    | [**TA-Lib**](https://mrjbq7.github.io/ta-lib/install.html) | 0.4.17   | 
+    | [**pandas**](https://pypi.org/project/pandas/)             | 1.1.1    | 
+    | [**numpy**](https://pypi.org/project/numpy/)               | 1.16.3   |
```
[the remark this answers is not in the document]
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/codeguide.md]: done

File: docs/pyalgotrad/structure.md (hunk 11 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/structure.md]: done

File: docs/pyalgotrad/introduction.md (hunk 10 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/introduction.md:15]: done

File: docs/pyalgotrad/codeguide.md (hunk 12 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/codeguide.md:30]: could not find the above comment in pyalgotrading

File: docs/pyalgotrad/codeguide.md [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+
+2. The strategy class should not import any packages/modules.
+3. The entire strategy should be a single Python class.
+
+---
+
+!!! Note
+    * Pandas is available with limited functionality. Direct printing of dataframes is disabled.
+    * It is a good idea to include the version number while installing a package using `pip`.
+    * Example - `pip install numpy==1.16.3`
```
[the remark this answers is not in the document]
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/codeguide.md:25]: done

File: docs/pyalgotrad/index.md (hunk 13 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done : added real trading again

File: docs/pyalgotrad/index.md (hunk 14 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/index.md]: done

File: docs/pyalgotrad/prerequisites.md (hunk 15 above)
Comment (prithviraj-gotepatil-algobulls) [reply docs/pyalgotrad/prerequisites.md]: done

**Review by pushpakdagade — APPROVED** [review]

**Review by pushpakdagade — APPROVED** [review]

**Pull request merged**